Sources/KeyboardSupport/KeyboardRespondable.swift

```diff
@@ -12,29 +12,33 @@ import UIKit
 
 /// Inherits from both KeyboardDismissable and KeyboardScrollable for convenience.
 public protocol KeyboardRespondable: KeyboardDismissable, KeyboardScrollable {
-    /// Must be called during setup ('viewDidLoad') so keyboard dismissal and responsiveness can be enabled.
-    func setupKeyboardRespondable()
+    /// Must be called during setup ('viewDidLoad') so keyboard dismissal and responsiveness can be enabled.  Returns gesture recognizer used for keyboard dismissal.
+    func setupKeyboardRespondable() -> UIGestureRecognizer
 }
 
 public extension KeyboardRespondable where Self: UIViewController {
-    func setupKeyboardRespondable() {
-        setupKeyboardDismissalView()
+    func setupKeyboardRespondable() -> UIGestureRecognizer {
+        return setupKeyboardDismissalView()
     }
 }
 
 // MARK: - KeyboardDismissable
 
 /// Enables automatic keyboard dismissal via tapping the screen when the keyboard is displayed.
 public protocol KeyboardDismissable: class {
-    /// Must be called once during setup ('viewDidLoad') to enable dismissal.
-    func setupKeyboardDismissalView()
+    /// Must be called once during setup ('viewDidLoad') to enable dismissal.  Returns gesture recognizer used for keyboard dismissal.
+    func setupKeyboardDismissalView() -> UIGestureRecognizer
 }
 
 public extension KeyboardDismissable where Self: UIViewController {
-    func setupKeyboardDismissalView() {
+    func setupKeyboardDismissalView() -> UIGestureRecognizer {
         let tapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(keyboardDismissalViewTapped))
         tapGestureRecognizer.cancelsTouchesInView = false
+        if #available(iOS 11.0, *) {
+            tapGestureRecognizer.name = String(describing: KeyboardDismissable.self)
```

Comment:
Would it make sense to use the conforming class here instead? I'm trying to think of a use case where you'd have multiple KeyboardDismissable views present in the same context.

Reply:
I agree, it might be better to be more specific. Something like:
```swift
tapGestureRecognizer.name = String(describing: "\(Self) - \(KeyboardDismissable.self)")
```

Reply:
I'm having a tough time thinking of a case for using more than one `KeyboardDismissable` in the the same context although it could happen. How about leaving the name up to the user?
```
func setupKeyboardDismissalView(gestureRecognizerName: String?) -> UIGestureRecognizer {
    let tapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(keyboardDismissalViewTapped))
    tapGestureRecognizer.cancelsTouchesInView = false
    if #available(iOS 11.0, *) {
        tapGestureRecognizer.name = gestureRecognizerName
    }
    view.addGestureRecognizer(tapGestureRecognizer)
    return tapGestureRecognizer
}
```

Reply:
My intention was to call out why/where the gesture recognizer was added.  If you used the conforming class instead of the protocol then the connection between the two would be obscured.  In an event where you had multiple KeyboardDismissable views, you can still differentiate between the gestures based on the view that they are attached to.

Reply:
alright, I removed the name